clients/hierarchy/hierarchy.go

```diff
@@ -67,38 +75,46 @@ func (c *Client) Healthcheck() (string, error) {
 }
 
 // GetRoot returns the root hierarchy response from the hierarchy API
-func (c *Client) GetRoot(instanceID, name string) (m Model, err error) {
+func (c *Client) GetRoot(ctx context.Context, instanceID, name string) (m Model, err error) {
 	path := fmt.Sprintf("/hierarchies/%s/%s", instanceID, name)
 
-	clientlog.Do(context.Background(), "retrieving hierarchy", service, path, log.Data{
+	clientlog.Do(ctx, "retrieving hierarchy", service, path, log.Data{
 		"method":      "GET",
 		"instance_id": instanceID,
 		"dimension":   name,
 	})
 
-	return c.getHierarchy(path)
+	return c.getHierarchy(path, ctx)
 }
 
 // GetChild returns a child of a given hierarchy and code
-func (c *Client) GetChild(instanceID, name, code string) (m Model, err error) {
+func (c *Client) GetChild(ctx context.Context, instanceID, name, code string) (m Model, err error) {
 	path := fmt.Sprintf("/hierarchies/%s/%s/%s", instanceID, name, code)
 
-	clientlog.Do(context.Background(), "retrieving hierarchy", service, path, log.Data{
+	clientlog.Do(ctx, "retrieving hierarchy", service, path, log.Data{
 		"method":      "GET",
 		"instance_id": instanceID,
 		"dimension":   name,
 		"code":        code,
 	})
 
-	return c.getHierarchy(path)
+	return c.getHierarchy(path, ctx)
 }
 
-func (c *Client) getHierarchy(path string) (m Model, err error) {
-	resp, err := c.cli.Get(c.url + path)
+func (c *Client) getHierarchy(path string, ctx context.Context) (m Model, err error) {
+
+	req, err := http.NewRequest("GET", c.url + path, nil)
 	if err != nil {
 		return
 	}
 
+	resp, err := c.cli.Do(ctx, req)
```

Comment:
I think you need to set the context on the request otherwise it will have its own which not contain any traceID's previously set.

```
req = req.WithContext(ctx)
```

Reply:
Ignore this - this is taken care of inside of `clienter.Do` 